Gothic-Unity-Project/Gothic-Unity
# Feature/75 async loading based on device refresh rate

## Description
Before this branch, we hard coded amount of world chunks + VOBs being created per frame. Now we dynamically decide based on device's fps settings when to skip to next frame. We also left all meshes visible during loading time which increased stress on rendering device. Meshes are all disabled now until game is fully loaded.

## To test
* Start Game on a MVR device (preferrably Quest 2)
* Check how fast the game loads the world
  * loading should be faster or at least comparable to loading from main branch
  * loading should have less hickups during loading

## My own test results:
* New logic
  * 72 fps on Quest 2 where we skip a frame after 50% threshold -> Game loaded in 14 sec
* Old logic:
  * Game loaded in 13 sec

Hint: Quest 2 is our lower end device. On a newer device, the game will load even faster as the loading chunks per frame are dynamically calculated
